cross-dev/src/main.rs

```diff
@@ -0,0 +1,208 @@
+#![deny(missing_debug_implementations, rust_2018_idioms)]
+
+use std::path::{Path, PathBuf};
+use std::process::{Command, Stdio};
+
+use clap::{Parser, Subcommand};
+use cross::CommandExt;
+use serde::Deserialize;
+
+// Store raw text data in the binary so we don't need a data directory
+// when extracting all targets, or running our target info script.
+const TARGET_INFO_SCRIPT: &str = include_str!("target_info.sh");
+const WORKFLOW: &str = include_str!("../../.github/workflows/ci.yml");
+
+#[derive(Parser, Debug)]
+#[clap(version, about, long_about = None)]
+struct Cli {
+    #[clap(subcommand)]
+    command: Commands,
+}
+
+#[derive(Subcommand, Debug)]
+enum Commands {
+    /// Extract and print info for targets.
+    ///
+    /// If no arguments are provided, this script will provide info
+    /// for all targets. The output will be similar to the following:
+    ///       | `aarch64-unknown-linux-musl`         | 1.2.0  | 9.2.0   | ✓   | 5.1.0 |
+    ///       | `i686-linux-android`                 | 9.0.8  | 9.0.8   | ✓   | 5.1.0 |
+    ///       | `i686-unknown-linux-musl`            | 1.2.0  | 9.2.0   | ✓   | N/A   |
+    ///       ...
+    #[clap(verbatim_doc_comment)]
+    TargetInfo {
+        /// If not provided, get info for all targets.
+        targets: Vec<String>,
+        /// Provide verbose diagnostic output.
+        #[clap(short, long)]
+        verbose: bool,
+        /// Image registry.
+        #[clap(long, default_value_t = String::from("ghcr.io"))]
+        registry: String,
+        /// Image repository.
+        #[clap(long, default_value_t = String::from("cross-rs"))]
+        repository: String,
+        /// Image tag.
+        #[clap(long, default_value_t = String::from("main"))]
+        tag: String,
+        /// Container engine (such as docker or podman).
+        #[clap(long)]
+        engine: Option<String>,
+    },
+}
+
+#[derive(Debug, PartialEq, Deserialize)]
+struct Workflow {
+    jobs: Jobs,
+}
+
+#[derive(Debug, PartialEq, Deserialize)]
+struct Jobs {
+    #[serde(rename = "generate-matrix")]
+    generate_matrix: GenerateMatrix,
+}
+
+#[derive(Debug, PartialEq, Deserialize)]
+struct GenerateMatrix {
+    steps: Vec<Steps>,
+}
+
+#[derive(Debug, PartialEq, Deserialize)]
+struct Steps {
+    env: Env,
+}
+
+#[derive(Debug, PartialEq, Deserialize)]
+struct Env {
+    matrix: String,
+}
+
+#[derive(Debug, PartialEq, Deserialize)]
+struct Matrix {
+    target: String,
+    os: String,
+    #[serde(default)]
+    cpp: i64,
+    #[serde(default)]
+    std: i64,
+    #[serde(default)]
+    run: i64,
+    runners: Option<String>,
```

Comment:
are these values used?

Reply:
They are not: I considered if they would and they should be removed.